### PR TITLE
install nvm/npm if one does not exist

### LINKED ISSUE #2: Directly use this feature without manually install npm breaks the installation procedure
In common devcontainer/devcontainer-contrib features, e.g. [jest](https://github.com/devcontainers-contrib/features/blob/006e18efa8d69a0e0097ada514e11449bfcd0ce6/src/jest/install.sh#L33), they check whether npm is installed and install one if needed (those codes are based on [feature template](https://github.com/devcontainers-contrib/features/blob/main/pkgs/feature-template/%7B%7Bcookiecutter.base_dir%7D%7D/src/%7B%7Bcookiecutter.feature_id%7D%7D/install.sh)).

Now unfortunately, when using a .devcontainer.json with only `image` and `features` specified, unfortunately (I don't know why) this feature is executed first, so no npm is installed and the whole procedure breaks.

I can also provide a PR.

### PR DESCRIPTION
**Pull request by yf-yang**
fix #2 
Note:
1. I am unable to run the tests until I change `test/pnpm` to `test/_global` and run the following command:
``` shell
devcontainer features test --global-scenarios-only .
```
Not sure if devcontainer cli changed, so I didn't commit that change. See the [code](https://github.com/devcontainers/feature-starter/blob/aac19152fc12052fa33f79e1041ae6b1b624c395/test/_global/color_and_hello.sh#L12).
2. Neither `installsAfter` nor `installAfter` actually works, see this [issue](https://github.com/devcontainers/spec/issues/160). Therefore I have to add `overrideFeatureInstallOrder` to make sure node is installed before pnpm. Otherwise, pnpm installation would be hided by node, which fails the test.

### CODE REVIEW

**Comment by NicoVIII** [comment]
Thank you very much!
This looks good to me, I will merge, fix the tests and release a new version shortly.

But I have a few follow-up questions, if I may:
The code is generated in a style which was formed by/for https://github.com/devcontainers-contrib/features, right? You also adjusted other things to match that style. Considering this, wouldn't it be a good idea to transfer the feature into that repository instead of having a different one just for pnpm in my user space?
One concern I have (which is not heavy right now) is, that the org devcontainers-contrib is currently controlled by only one user, which would be a single-point-of-failure for a community thing like this. That's why I would like to have your opinion on that, because you seem to have some experience with community and project :)

**Comment by yf-yang** [comment]
:smile: Actually I was just using vscode's "Create Dev Container" prompt and found your pnpm feature, but I didn't check how those features were registered in vscode's prompt (although most of them were from devcontainer-contrib). Therefore, I simply assumed there were some standard procedure to register features in devcontainer-contrib and pnpm was not merged yet.

Anyway, I think SPOF is not that bad, because we are treating features independently, so although we can make them decentralized, each one would still be centralized. Moreover, those features are following the same template, so it is much easier to navigate to potential bugs and fix all of them in a similar way. If the maintainer stops to maintain the codebase, someone could easily fork it and continue the work, it is better than having multiple repos and you will never know when someone would stop maintaining.

To summarize, my point is, try to make them stay in one repo, the count of actual maintainers doesn't matter.

**Comment by yf-yang** [comment]
Besides, does `installsAfter` work? That should be the right name but I didn't see it work.

**Comment by NicoVIII** [comment]
As far, as I could see it works... sometimes. In my experience it works, if you don't specify a version on the other package in the feature list. Which is unfortunate, because this seems to be the default way to specify those.

I reported that already, but sadly the issue was closed:
https://github.com/devcontainers/cli/issues/269
